builtin/logical/kv/path_metadata.go

```diff
@@ -81,6 +82,36 @@ version-agnostic information about a secret.
 	}
 }
 
+// pathDetailedMetadata returns the path configuration for LIST operations on
+// the detailed metadata endpoint (with metadata included in responses).
+func pathDetailedMetadata(b *versionedKVBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: "detailed-metadata/" + framework.MatchAllRegex("path"),
+		Fields: map[string]*framework.FieldSchema{
+			"path": {
+				Type:        framework.TypeString,
+				Description: "Location of the secret.",
+			},
+			"after": {
+				Type:        framework.TypeString,
+				Description: `Optional entry to list begin listing after, not required to exist. Only used for listing.`,
+			},
+			"limit": {
+				Type:        framework.TypeInt,
+				Description: `Optional number of entries to return; defaults to all entries. Only used for listing.`,
+			},
+		},
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ListOperation: b.upgradeCheck(b.pathMetadataList()),
+		},
+
+		ExistenceCheck: b.metadataExistenceCheck(),
+
+		HelpSynopsis:    confHelpSyn,
```

Comment:
Interestingly, this is a shortcoming with the existing `pathMetadata` as well; probably needs to be fixed in a follow-up PR.